MAINT: testing syntax

jklenzing opened the pull request:
# Description

Addresses pysat/pysat#1053, pysat/pysat#1042, pysat/pysat#1048

- Updates testing syntax to maintain pytest standards. 
- Updates Github Actions syntax
- Removes nose requirements
- restores changelog from previous version
- Adds RC workflow

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Via Github Actions

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

 If this is a release PR, replace the first item of the above checklist with the
 release checklist on the pysat wiki:
 https://github.com/pysat/pysat/wiki/Checklist-for-Release

jklenzing committed:
MAINT: pytest syntax

jklenzing committed:
DOC: update changelog

jklenzing committed:
TST: update testing setup

jklenzing committed:
MAINT: update doc tests

jklenzing committed:
BUG: fix nep29

aburrell commented on CHANGELOG.md:
```diff
@@ -13,4 +13,21 @@ This project adheres to [Semantic Versioning](https://semver.org/).
   * Adopted latest pysat development standards.
+  * Adopt pytest syntax
 
+[0.3.2] - 2022-05-13
+--------------------
+* New Features
+  * Compatible with pysat v3.0+
+* Deprecations
+* Documentation
+  * Added pull request templates and other GitHub project documentation.
+  * Switched Windows installation instructions to favor installing WSL.
+* Bug Fix
+  * Improved builds for newer compilers.
+  * Replaces uninterpretable characters with '*' so data loading may continue.
+* Maintenance
+  * Adopted latest pysat development standards.
+  * Shifted from TravisCI to GitHub Actions for online testing.
+  * Adopted setup.cfg
+  * Improved PEP8 compliance
```
This contains a lot more changes than I see in this pull request.

jklenzing replied:
During a previous pull, the changelog was apparently deleted for previous versions.  restores based on release notes.

jklenzing replied:
merged changes from main into develop-3.  This bit still shows up as a duplicate since it's a separate commit.

jklenzing committed:
Update test_requirements.txt

jklenzing committed:
Merge branch 'develop-3' into tst/syntax

jklenzing commented:
Added a flake-8 cap to this branch since the latest version does not like how we input exceptions.  Documented as part of standards upgrade in #43.

jklenzing committed:
Create pysat_rc.yml

jklenzing commented on .github/workflows/pysat_rc.yml:
```diff
@@ -0,0 +1,51 @@
+
+name: Test with latest pysat RC
+
+on: [push]
```
```suggestion
on: [workflow_dispatch]
```

jklenzing committed:
Update .github/workflows/pysat_rc.yml

jklenzing commented:
Went ahead and added the RC check to close out issues at pysat.  Results of test are here: https://github.com/pysat/pysatCDF/actions/runs/3594442005

Added #44 to capture the updating Github Actions to include windows and mac tests.  Probably best for these to be implemented after or during meson upgrades to the install procedure.

rstoneback commented:
Thanks @jklenzing and @aburrell!
